Proposal for supporting trunked port environments

jacobweinstock opened the pull request:
## Description

<!--- Please describe what this PR is going to change -->

Add support for trunked port environments.
Rendered proposal: https://github.com/jacobweinstock/proposals/blob/vlan-support/proposals/0030/README.md 

## Why is this needed

<!--- Link to issue you have raised -->

Fixes: #

## How Has This Been Tested?
<!--- Please describe in detail how you tested your changes. -->
<!--- Include details of your testing environment, and the tests you ran to -->
<!--- see how your change affects other areas of the code, etc. -->


## How are existing users impacted? What migration steps/scripts do we need?

<!--- Fixes a bug, unblocks installation, removes a component of the stack etc -->
<!--- Requires a DB migration script, etc. -->


## Checklist:

I have:

- [ ] updated the documentation and/or roadmap (if required)
- [ ] added unit or e2e tests
- [ ] provided instructions on how to upgrade


-----
[View rendered proposals/0030/README.md](https://github.com/jacobweinstock/proposals/blob/vlan-support/proposals/0030/README.md)

chrisdoherty4 reviewed (dismissed):
Reviewed a while back. Not sure why I didn't approve. Looks great!

displague commented on proposals/0028/README.md:
```diff
@@ -0,0 +1,121 @@
+
+## Alternatives
+
+Don't support VLAN tagging.
```
One of the items discussed in the community calls was wether or not there was an existing standard for DHCP declaring VLAN configuration. @jacobweinstock pointed out that there were none and that the nearest results were vendor specific and tailored to VOIP solutions (some using option 43, others 132):

https://www.iana.org/assignments/bootp-dhcp-parameters/bootp-dhcp-parameters.xhtml
> Option 132 - IEEE 802.1Q VLAN ID

Here are some examples of vendors providing or expecting VLAN designation features over DHCP Options:
* https://www.sonicwall.com/support/knowledge-base/how-to-configure-dhcp-option-132-to-get-dhcp-lease-from-vlan/220309054946100/
* https://wiki.unify.com/wiki/VLAN_ID_Discovery_over_DHCP#Example_configuration_of_an_Cisco_Switch
* https://www.teradici.com/web-help/TER1206003/4.x-4.9/07_HowTo/Config_VLAN.htm
* https://wiki.freepbx.org/display/PHON/DHCP+VLAN+Option+132

jacobweinstock replied:
Hey @displague , thanks for this detail! I've updated the proposal to include this info.

jacobweinstock committed:
Move to proposal 0030:

Another proposal was merged as 0028, so I
moved this one to 0030. Also, added a comment
regarding DHCP option 132.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>

jacobweinstock committed:
Add more detail around option 132 vs option 43:

Hopefully this will help folks that want to dig
into this option 132 a little more.

Signed-off-by: Jacob Weinstock <jakobweinstock@gmail.com>
